advanced/button-interrupt/solution/src/main.rs

```diff
@@ -44,44 +29,46 @@ fn main() -> anyhow::Result<()> {
         intr_type: gpio_int_type_t_GPIO_INTR_POSEDGE, // positive edge trigger = button down
     };
 
-    let (mut tx, rx) = mpsc::channel::<()>();
+    // Queue configurations
     const QUEUE_TYPE_BASE: u8 = 0;
     const ITEM_SIZE: u32 = 0; // we're not posting any actual data, just notifying
-    const QUEUE_SIZE: u32 = 2; // 1 might be enough?
-    unsafe {
-        // ISR wakes an xQueue, which in turn communicates with main() using an `mpsc::channel`.
-        // TODO: figure out how to use channel directly in ISR - currently panics inside `lock_acquire_generic`.
-        // probable cause: fake atomics
-        // `mpsc::Sender` uses atomics. Atomics decompose to `__atomic_store_4`. That function calls `vPortSetInterruptMask`.
-        EVENT_QUEUE = Some(xQueueGenericCreate(QUEUE_SIZE, ITEM_SIZE, QUEUE_TYPE_BASE));
-
-        let task_name = CString::new("button_task")?;
-        const NO_AFFINITY: i32 = 0x7FFFFFFF;
-        xTaskCreatePinnedToCore(
-            Some(button_task),
-            task_name.as_c_str().as_ptr(),
-            2048,
-            &mut tx as *mut _ as *mut c_void,
-            10,
-            ptr::null_mut(),
-            NO_AFFINITY,
-        );
+    const QUEUE_SIZE: u32 = 1;
 
+    unsafe {
+        // Writes button configuration into register
         esp!(gpio_config(&io_conf))?;
-
+        
+        // The flag that the interrupt is handled in RAM
         const ESP_INTR_FLAG_IRAM: i32 = 1 << 10; // ISR will be executed even when caches are disabled
```

Comment:
This is in [`esp-idf-sys`](https://esp-rs.github.io/esp-idf-sys/esp_idf_sys/constant.ESP_INTR_FLAG_IRAM.html) so we can import it rather than define it.